yt_dlp/extractor/gofile.py

```diff
@@ -0,0 +1,65 @@
+# coding: utf-8
+from .common import InfoExtractor
+from ..utils import (
+    ExtractorError,
+    try_get
+)
+
+
+class GofileIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?gofile\.io/d/(?P<id>[^/]+)'
+    _TESTS = [{
+        'url': 'https://gofile.io/d/AMZyDw',
+        'info_dict': {
+            'id': 'AMZyDw',
+        },
+        'playlist_mincount': 2,
+        'playlist': [{
+            'info_dict': {
+                'id': 'de571ac1-5edc-42e2-8ec2-bdac83ad4a31',
+                'filesize': 928116,
+                'ext': 'mp4',
+                'title': 'nuuh'
+            }
+        }]
+    }]
+
+    def _entries(self, file_id):
+
+        # Create guest account
+        accountdata = self._download_json('https://api.gofile.io/createAccount', 'Gofile', note='Getting a new guest account')
+        accountToken = accountdata['data']['token']
+
+        # Get file list
+        requesturl = f'https://api.gofile.io/getContent?contentId={file_id}&token={accountToken}&websiteToken=websiteToken&cache=true'
+        filelist = self._download_json(requesturl, 'Gofile', note='Getting filelist')
+
+        status = filelist['status']
+        if status != "ok":
+            raise ExtractorError('Received error from service, status: %s\n' % status, expected=True)
+
+        contents = try_get(filelist, lambda x: x['data']['contents'], dict)
+
+        for _, file in contents.items():
+
+            filetype = file['mimetype'].split("/", 1)[0]
+            if filetype != "video" and filetype != "audio":
+                continue
+
+            filedata = {
+                'id': file['id'],
+                'title': file['name'].rsplit('.', 1)[0],
+                'url': file['directLink'],
+                'filesize': file['size'],
+                'release_timestamp': file['createTime']
+            }
+            yield filedata
```

Comment:
The only things we absolutely need are id, title, URL. The code must tolerate failure in getting the other fields. Use `.get`/`try_get` as necessary